### DISCUSSION TITLE
Instalation error

### DISCUSSION THREAD

**Discussion by RosnerTech** [requarks/wiki#5483] [Help / Questions] [answered]
Dear,

I am installing version v2.5.285, using the procedure https://docs.requarks.io/install/linux, after all
the procedures when executing the node server command shows an error: Package subpath './public/extractFiles' is not defined by "exports" in /vr/wiki/node_modules/extract-files/package.json.
Could you help me with this situation?

erro

Att.

Rosner

**Comment by andrewzah** [comment]
I have the same issue.

**Comment by NGPixel** [comment]
Which version of Node.js are you using?

**Reply by RosnerTech** [reply]
Hello,

I'm using Node v18.2.0 on Debian 10.

**Reply by NGPixel** [reply, marked as answer]
That's not supported. Please read the [requirements](https://docs.requarks.io/install/requirements#nodejs).

**Reply by 1Luc1** [reply]
I have the same issue on Node v18.6.0 on Debian 11.

~~Are there any plans to support node18 in the future?~~
Just found out that node 18 will be supported with Wiki.js v3.0 .
So I guess, I have to wait.

**Comment by 1Luc1** [comment]
I digged in a little deeper and found an temporary solution for me.

Added the following lines to the `exports` section of the file `node_modules/extract-files/package.json`
```
"./public/extractFiles": "./public/extractFiles.js",
"./public/isExtractableFile": "./public/isExtractableFile.js",
"./public/ReactNativeFile": "./public/ReactNativeFile.js",
```
**related links**
https://github.com/jaydenseric/extract-files/issues/17
https://nodejs.org/api/packages.html#packages_subpath_patterns

**EDIT:**
As of the latest version v2.5.299 and this commit https://github.com/requarks/wiki/commit/e1d282ad1102fb298528238d3c6f18f073dbeb7d you better comment out the unsupported nodejs version checker.

**Reply by RosnerTech** [reply]
thank you,
I will apply in my environment.

Att.
Rosner

**Reply by Tedbudd3** [reply]
Could you please explain where in the json file you added these lines?

**Reply by 1Luc1** [reply]
within the `exports` section ...
mine looks something like this
```json
"exports": {
    ".": {
      "import": "./public/index.mjs",
      "require": "./public/index.js"
    },
    "./public/extractFiles": "./public/extractFiles.js",
    "./public/isExtractableFile": "./public/isExtractableFile.js",
    "./public/ReactNativeFile": "./public/ReactNativeFile.js",
    "./public/": "./public/",
    "./package": "./package.json",
    "./package.json": "./package.json"
  },

```

Nevertheless, I don't know if this is still necessary since the supported nodejs versions were updated -> https://docs.requarks.io/install/requirements#nodejs

**Reply by Tedbudd3** [reply]
Thank you very much for your reply. We (as a company) running wiki.js v2.5.297 and node.js v16.20.0, for a vulnerability remediation, I have to upgrade node.js to a newer version, I tried last week upgrading to nodejs version 20، didn’t work after running the node server powershell command, do you have any ideas on what could be the problem? Not sure if I was supposed to upgrade to version 18 instead or even maybe upgrade the wiki.js. I would really appreciate your help with this issue.

**Reply by 1Luc1** [reply]
I doubt this is the right place for this conversation and you may ask this in some other, more help centered environment (e.g. [stackoverflow](https://stackoverflow.com/)). And also provide some more detailed description / error logs of what exactly is not working.

**Comment by TimHoepfner** [comment]
Same for me, thanks @1Luc1
